db/migrate/20250912000001_populate_pulp_prn_fields.rb

```diff
@@ -0,0 +1,387 @@
+        WHEN publication_href ~ '/publications/container/container/' THEN
+          'prn:container.containerpublication:' ||
+          regexp_replace(publication_href, '^.*/([^/]+)/?$', '\\1')
+        WHEN publication_href ~ '/publications/ansible/ansible/' THEN
+          'prn:ansible.ansiblepublication:' ||
+          regexp_replace(publication_href, '^.*/([^/]+)/?$', '\\1')
+        WHEN publication_href ~ '/publications/python/pypi/' THEN
+          'prn:python.pythonpublication:' ||
+          regexp_replace(publication_href, '^.*/([^/]+)/?$', '\\1')
+        WHEN publication_href ~ '/publications/file/file/' THEN
+          'prn:file.filepublication:' ||
+          regexp_replace(publication_href, '^.*/([^/]+)/?$', '\\1')
+        ELSE NULL
+      END"
+    )
+
+    # Ansible Collections
+    Katello::AnsibleCollection.where.not(pulp_id: nil).update_all(
+      "pulp_prn = CASE
+        WHEN pulp_id ~ '/content/ansible/collection_versions/' THEN
+          'prn:ansible.collectionversion:' ||
+          regexp_replace(pulp_id, '^.*/([^/]+)/?$', '\\1')
+        ELSE NULL
+      END"
+    )
+
+    # Generic Content Units
+    Katello::GenericContentUnit.where.not(pulp_id: nil).update_all(
+      "pulp_prn = CASE
+        WHEN pulp_id ~ '/content/python/packages/' THEN
+          'prn:python.pythonpackagecontent:' ||
+          regexp_replace(pulp_id, '^.*/([^/]+)/?$', '\\1')
+        WHEN pulp_id ~ '/content/ostree/refs/' THEN
+          'prn:ostree.ostreeref:' ||
+          regexp_replace(pulp_id, '^.*/([^/]+)/?$', '\\1')
+        ELSE NULL
+      END"
+    )
+
+    # Module Streams
+    Katello::ModuleStream.where.not(pulp_id: nil).update_all(
+      "pulp_prn = CASE
+        WHEN pulp_id ~ '/content/rpm/modulemds/' THEN
+          'prn:rpm.modulemd:' ||
+          regexp_replace(pulp_id, '^.*/([^/]+)/?$', '\\1')
+        ELSE NULL
+      END"
+    )
+
+    # Docker Manifest Lists
+    Katello::DockerManifestList.where.not(pulp_id: nil).update_all(
+      "pulp_prn = CASE
+        WHEN pulp_id ~ '/content/container/manifests/' THEN
+          'prn:container.manifest:' ||
+          regexp_replace(pulp_id, '^.*/([^/]+)/?$', '\\1')
+        ELSE NULL
+      END"
+    )
+
+    # Docker Manifests
+    Katello::DockerManifest.where.not(pulp_id: nil).update_all(
+      "pulp_prn = CASE
+        WHEN pulp_id ~ '/content/container/manifests/' THEN
+          'prn:container.manifest:' ||
+          regexp_replace(pulp_id, '^.*/([^/]+)/?$', '\\1')
+        ELSE NULL
+      END"
+    )
+
+    # Docker Tags
+    Katello::DockerTag.where.not(pulp_id: nil).update_all(
+      "pulp_prn = CASE
+        WHEN pulp_id ~ '/content/container/tags/' THEN
+          'prn:container.tag:' ||
+          regexp_replace(pulp_id, '^.*/([^/]+)/?$', '\\1')
+        ELSE NULL
+      END"
+    )
+
+    # RPMs - Process in batches due to potentially millions of records
+    Katello::Rpm.where.not(pulp_id: nil).in_batches(of: 10_000) do |batch|
```

Comment:
What do you think about having bulk processing on all content units?

Reply:
I could see someone having a ton of container image tags & manifests, for example.

Reply:
Since this is all in the katello DB, processing should be fairly quick..In batches of 10,000..Also reduces memory overhead if we do it in batches rather than all at once..